lib/livebook/runtime/evaluator.ex

```diff
@@ -624,6 +626,11 @@ defmodule Livebook.Runtime.Evaluator do
     |> Map.update!(:context_modules, &(&1 ++ prev_env.context_modules))
   end
 
+  # TODO: Make sure to change this into a folder which pertains the session.
+  defp tmp_filename() do
+    :lists.flatten(:io_lib.format("/tmp/erlang-eval-~p.tmp", [:erlang.phash2(make_ref())]))
+  end
```

Comment:
We get `tmp_dir` on init, so we could use that as the base. But let's try the fileless approach first.

Reply:
The fileless approach won't work.

@fnchooft we pass a `tmp_dir` to init:

https://github.com/livebook-dev/livebook/blob/62d8c4e0e926005d9440a4b97cd6a673fc189eb4/lib/livebook/runtime/evaluator.ex#L276

Can you store it as part of the state and use it to build the tmp directory? Notice that it may be `nil` in case of read-only file systems, so we should return an error in those cases saying you can't define erlang modules in read-only file systems. :)